pkg/gator/reader/read_constraints.go

```diff
@@ -1,6 +1,7 @@
 package reader
```

Comment:
Overall generally LGTM but needs unit tests.

Some things we should test:

1. converting a "compact representation" to an "expended representation".  The pure function will make this test clearer.
2. converting actual text on a fake template into a full SyncRequirements
3. Some error scenarios where the annotation is malformed (someone's bound to mess up the formatting, particularly if the annotation picks up steam in the community)

Reply:
Thanks! Added tests--I think adding tests for ExpandCompactRequirements would be pretty redundant but lmk what you think